### ISSUE TITLE
CSV files, variable number of fields

### ISSUE THREAD

**Issue by PSLLSP** [simonmichael/hledger#2188]
hledger 1.32.3

Current `hledger` can process only well formed CSV files. It can skip several lines at the start of the file. And a rule `end` can be used to force skip to the rest of file. 

I would like to have an option to process CSV files with variable number of fields, like that when field is missing, `hledger` will not end with an error hen field is missing

* there could be an option to define minimal number of mandatory fields
* new variable will be defined, that will keep number of fields in the current record; `awk` has variable `NF`, something like that

There is rule `skip` in  `if block` statement, this could be used to skip records that doesn't have correct number of fields or take an action on these records, like adding missing information.

---

I have CSV records, those have several header lines followed by CSV records those have data I want to process. I can design hledger rule file that process one such file but I cannot design rule file that process those files when joined with `cat *.csv > import.csv`.

Example:

```
$ cat import.csv
statement: 2023/01
account: 123456789/4321
owner: John Smith
start date: 1.1.2023
end date: 31.1.2023
opening balance:  100.00 EUR
closing balance:  100.50 EUR

"ID";"Date";"Volume","Currency";"Description","Note"
"54321";"31.01.2023";"0.5";"EUR";Interest";""

statement: 2023/02
account: 123456789/4321
owner: John Smith
start date: 1.2.2023
end date: 28.2.2023
opening balance:  100.50 EUR
closing balance:  101.00 EUR

"ID";"Date";"Volume","Currency";"Description","Note"
"56324";"28.02.2023";"0.5";"EUR";Interest";""
```

I would like to define that no fields are mandatory (to be able to skip empty lines)
I would like to use `if block` rule to:

* skip records/lines those do not have 6 fields
* skip records where field1 (`%1`) has value `ID` - to ignore headers
* process the rest like regular CSV records

**Comment by Thaodan** [comment]
Is it possible to check the amount of column's in a CSV file? At least for dealing with older/newer exports it could be useful.

**Comment by simonmichael** [comment]
Yes, you can use a conditional's regexp to count the separator character and apply different rules based on the number of fields. Eg, for comma-separated data:
```
# Skip section 1's 6-field lines
if ^([^,]*,){6}$
 skip
```

I think you could also name all possible fields and then select based on whether rightmost fields contain a value.

**Comment by Thaodan** [comment]
Does that also work with field assignments? I.e. if a newer format
contains an extra field that wasn't present before.

**Comment by simonmichael** [comment]
I'm not sure - try it and let us know ?
